linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -785,13 +829,15 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
    *  a method comment). However, the global state modifications are
    *  concurrency safe.
    */
-  private[optimizer] abstract class MethodImpl(val owner: MethodContainer,
+  private[optimizer] final class MethodImpl(owner: MethodContainer,
       val methodName: MethodName)
       extends OptimizerCore.MethodImpl with OptimizerCore.AbstractMethodID
       with Unregisterable {
 
     private[this] var _deleted: Boolean = false
 
+    private val bodyAskers = CollOps.emptyMap[MethodImpl, Unit]
+    private val registeredTo = CollOps.emptyMap[Unregisterable, Unit]
```

Comment:
`registeredTo` was previously an `AtomicAcc[Unregisterable], i.e., an `Addable[Unregisterable]`. Shouldn't we keep using the `Addable` abstraction during this refactoring?

Reply:
Well, the `Addable` implements parallel traversal which comes at a cost of copying before traversing. So I'm not sure that is better. Also, it seems to me that `registeredTo` really should be a `Set` anyways...

Reply:
Right, that makes sense.